application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelMonitor.java

```diff
@@ -0,0 +1,245 @@
+import java.util.stream.Stream;
+
+
+/**
+ * A class responsible for monitoring the status of channels and reporting on their busy/free status
+ * for use by {@link FreeCommand}
+ */
+public class ChannelMonitor {
+    // Map to store channel ID's, use Guild.getChannels() to guarantee order for display
+    private final Map<Long, ChannelStatus> channelsToMonitor;
+    private final Map<Long, Long> postStatusInChannel;
+
+
+
+    ChannelMonitor() {
+        postStatusInChannel = new HashMap<>(); // JDA required to fetch guildID
+        channelsToMonitor = new HashMap<>();
+    }
+
+    /**
+     * Method for adding channels that need to be monitored.
+     *
+     * @param channelId the id of the channel to monitor
+     */
+    public void addChannelToMonitor(final long channelId) {
+        channelsToMonitor.put(channelId, new ChannelStatus(channelId));
+    }
+
+    /**
+     * Method for adding the channel that the status will be printed in. Even though the method only
+     * stores the long id it requires, the method requires the actual {@link TextChannel} to be
+     * passed because it needs to verify it as well as store the guild id.
+     *
+     * This method also calls a method which updates the status of the channels in the {@link Guild}
+     * . So always add the status channel AFTER you have added all monitored channels for the guild,
+     * see {@link #addChannelToMonitor(long)}.
+     *
+     * @param channel the channel the status message must be displayed in
+     */
+    public void addChannelForStatus(@NotNull final TextChannel channel) {
+        postStatusInChannel.put(channel.getGuild().getIdLong(), channel.getIdLong());
+        updateStatusFor(channel.getGuild());
+    }
+
+    /**
+     * This method tests whether a guild id is configured for monitoring in the free command system.
+     * 
+     * @param guildId the id of the guild to test.
+     * @return {@code true} if the guild is configured in the system, {@code false} otherwise.
+     */
+    public boolean isMonitoringGuild(final long guildId) {
+        return postStatusInChannel.containsKey(guildId);
+    }
+
+    /**
+     * This method tests whether a channel id is configured for monitoring in the free command
+     * system.
+     * 
+     * @param channelId the id of the channel to test.
+     * @return {@code true} if the channel is configured in the system, {@code false} otherwise.
+     */
+    public boolean isMonitoringChannel(final long channelId) {
+        return channelsToMonitor.containsKey(channelId);
+    }
+
+    /**
+     * This is a delegation method to pass operations to {@link ChannelStatus}. see
+     * {@link ChannelStatus#isBusy()} for details.
+     *
+     * @param channelId the id for the channel to test.
+     * @return {@code true} if the channel is 'busy', false if the channel is 'free'.
+     */
+    public boolean isChannelBusy(final long channelId) {
+        return channelsToMonitor.get(channelId).isBusy();
```

Comment:
Shouldn't we use `Objects#requireNonNull`?

Reply:
no we shouldnt

if u have read any of the discussions regarding `requireNonNull` on other parts of this bot u would know that requireNonNull is for validating input parameters to create fail fast. so that the exception is thrown where it is generated instead of a dozen method calls later. This will fail fast atm, because the possible null is immediately called. 

To change this to an explicit exception throw u would not use NPE because it doesnt make sense in this context, this would be an IllegalArgumentException. another option is using optional and another option is chaining isChannelMonitored() call before returning a result because it throws an exception.

Reply:
> 
> 
> Shouldn't we use `Objects#requireNonNull`?

fyi

Objects#requireNonNull throws a NPE in it's code.
So if we used Objects#requireNonNull it'd only add more code, and worse debugging experience.

If we don't validate anything, Java throws an NPE that's the channel is null, instead of a NPE without any info.

Reply:
fixed